conform-pr/main.go

```diff
@@ -28,52 +28,70 @@ func main() {
 		action.Fatalf("main: %s", err)
 	}
 
-	errors := runChecks(action, client)
-
-	if len(errors) == 0 {
-		return
+	summaries := runChecks(action, client)
+	action.AddStepSummary("| Check  | Status |")
+	action.AddStepSummary("|--------|--------|")
+
+	for _, summary := range summaries {
+		statusSign := ":apple:"
+		if summary.Ok {
+			statusSign = ":green_apple:"
+		}
+		if summary.Details != nil {
+			action.AddStepSummary(fmt.Sprintf("|%s | %s %s|", summary.Name, statusSign, summary.Details))
+		} else {
+			action.AddStepSummary(fmt.Sprintf("|%s | %s |", summary.Name, statusSign))
+		}
 	}
 
-	var serrors []string
-	for _, err := range errors {
-		serrors = append(serrors, fmt.Sprintf("%s", err))
+	for _, v := range summaries {
+		if !v.Ok {
+			action.Fatalf("The PR does not conform to the rules")
+		}
 	}
+}
 
-	action.Fatalf("The PR does not conform to the rules:\n - %s.", strings.Join(serrors, ";\n - "))
+type Summary struct {
+	Name    string
+	Ok      bool
+	Details error
 }
 
 // runChecks runs all the checks included into the PR conformance rules.
-// It returns the list of errors that occurred during the checks.
-func runChecks(action *githubactions.Action, client graphql.Querier) []error {
-	var errors []error
-
+// It returns the list of check summary for the checks.
+func runChecks(action *githubactions.Action, client graphql.Querier) []Summary {
 	pr, err := getPR(action, client)
 	if err != nil {
-		return []error{fmt.Errorf("runChecks: %w", err)}
+		return []Summary{{Name: "Read PR", Details: err}}
 	}
 
 	// PRs from dependabot are perfect
 	if pr.author == "dependabot[bot]" {
 		return nil
 	}
 
-	if err := pr.checkTitle(); err != nil {
-		errors = append(errors, err)
+	titleSummary := Summary{Name: "Title"}
+	titleSummary.Details = pr.checkTitle()
+	if titleSummary.Details == nil {
+		titleSummary.Ok = true
 	}
 
-	if err := pr.checkBody(action); err != nil {
-		errors = append(errors, err)
+	bodySummary := Summary{Name: "Body"}
+	bodySummary.Details = pr.checkBody(action)
+	if bodySummary.Details == nil {
+		bodySummary.Ok = true
 	}
 
-	return errors
+	return []Summary{titleSummary, bodySummary}
 }
 
-// getPR returns PR's information.
-// If an error occurs, it returns nil and the error.
+// getPR returns PR's information and returns
+// * pull request details if no errors occured
+// * a summary list whether and which check passed successfully or not.
```

Comment:
That's no longer true. Let's revert this change

Reply:
done.